homeassistant/components/starlink/sensor.py

```diff
@@ -0,0 +1,125 @@
+"""Contains sensors exposed by the Starlink integration."""
+from __future__ import annotations
+
+from datetime import datetime, timedelta
+
+from homeassistant.components.sensor import SensorDeviceClass, SensorStateClass
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import DATA_RATE_MEGABITS_PER_SECOND, DEGREE, TIME_MILLISECONDS
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .entity import StarlinkSensorEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up all sensors for this entry."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+
+    async_add_entities(
+        [
+            PingSensor(coordinator=coordinator),
+            AzimuthSensor(coordinator=coordinator),
+            ElevationSensor(coordinator=coordinator),
+            UplinkThroughputSensor(coordinator=coordinator),
+            DownlinkThroughputSensor(coordinator=coordinator),
+            LastBootTime(coordinator=coordinator),
+        ]
+    )
+
+
+class PingSensor(StarlinkSensorEntity):
+    """A sensor exposing Dishy's latency in milliseconds."""
+
+    _attr_state_class = SensorStateClass.MEASUREMENT
+    _attr_native_unit_of_measurement = TIME_MILLISECONDS
+    _attr_icon = "mdi:speedometer"
+    _attr_unique_id = "ping"
```

Comment:
I would suggest not using attributes of a base class for non-final values. Basically, when you see `_attr_unique_id` you expect it to be THE unique ID of an entity, not something that is used to construct it somewhere else.
Again, if you use EntityDescription this will become `entity_description.key`